fetch_recovered_data.py

```diff
@@ -0,0 +1,72 @@
+import argparse
+import csv
+from datetime import datetime
+from io import StringIO
+
+import pytz
+from core_data_modules.logging import Logger
+from core_data_modules.traced_data import Metadata, TracedData
+from core_data_modules.traced_data.io import TracedDataJsonIO
+from core_data_modules.util import TimeUtils, SHAUtils, IOUtils
+from storage.google_cloud import google_cloud_utils
+
+from src.lib import PipelineConfiguration
+
+log = Logger(__name__)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Converts messages in de-identified recovery CSV format to TracedData "
+                                                 "which can be processed by the pipeline")
+
+    parser.add_argument("user", help="Identifier of the user launching this program")
+    parser.add_argument("google_cloud_credentials_file_path", metavar="google-cloud-credentials-file-path",
+                        help="Path to a Google Cloud service account credentials file to use to access the "
+                             "credentials bucket")
+    parser.add_argument("pipeline_configuration_file_path", metavar="pipeline-configuration-file",
+                        help="Path to the pipeline configuration json file")
+    parser.add_argument("raw_data_dir", metavar="raw-data-dir",
+                        help="Path to a directory to save the raw data to")
+
+    args = parser.parse_args()
+
+    user = args.user
+    google_cloud_credentials_file_path = args.google_cloud_credentials_file_path
+    pipeline_configuration_file_path = args.pipeline_configuration_file_path
+    raw_data_dir = args.raw_data_dir
+
+    # Load the pipeline configuration file
+    log.info("Loading Pipeline Configuration File...")
+    with open(pipeline_configuration_file_path) as f:
+        pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+
+    for recovery_csv_url in pipeline_configuration.recovery_csv_urls:
+        log.info(f"Downloading recovered data from '{recovery_csv_url}'...")
+        raw_csv_string = StringIO(google_cloud_utils.download_blob_to_string(
+            google_cloud_credentials_file_path, recovery_csv_url))
+        raw_data = list(csv.DictReader(raw_csv_string))
+        log.info(f"Downloaded {len(raw_data)} recovered messages")
+
+        log.info("Converting the recovered messages to TracedData...")
+        data = []
+        for i, row in enumerate(raw_data):
+            raw_date = row["Received On"]
+            parsed_raw_date = datetime.strptime(raw_date, "%m/%d/%y %H:%M")
+            localized_date = pytz.timezone("Africa/Mogadishu").localize(parsed_raw_date)
+
```

Comment:
Please can you add an assersion that `row["Mobile No"]` contains something that looks like an avf id - e.g. starts with avf-....

Reply:
added